### PR TITLE
The LDM decided to try : instead of => in the switch expression syntax.

### LINKED ISSUE #26514: Change `=>` to `:` in switch expression prototype
The LDM decided to try `:` instead of `=>` in the switch expression syntax. Change the prototype to accept `:` instead of `=>`.

### PR DESCRIPTION
**Pull request by gafter**
This changes the prototype to accept : instead of =>
Fixes #26514

@cston @agocke @jcouv Could you please review this change for Build?
@dotnet/roslyn-compiler

### CODE REVIEW

File: src/Compilers/CSharp/Test/Syntax/Parsing/PatternParsingTests.cs [hunk 1]
```diff
@@ -2372,32 +2378,24 @@ public void SwitchExpression02()
...
-                            }
+                            N(SyntaxKind.IdentifierToken, "c");
                         }
                     }
                 }
-                N(SyntaxKind.CloseBraceToken);
+                M(SyntaxKind.CloseBraceToken);
             }
             EOF();
         }
```
Comment (cston) [inline src/Compilers/CSharp/Test/Syntax/Parsing/PatternParsingTests.cs]: >} [](start = 8, length = 1)

Perhaps test `switch { a when b ? c : d : e }`.

File: src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs [hunk 2]
```diff
@@ -682,9 +682,9 @@ private SeparatedSyntaxList<SwitchExpressionArmSyntax> ParseSwitchExpressionArms
                 // expression is not permitted as a constant expression here; it would have to be parenthesized.
```
[the remark this answers is not in the document]
Comment (gafter) [reply src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs]: No. We should accept a ternary expression now. I will revise the PR.

---
In reply to: [185150957](https://github.com/dotnet/roslyn/pull/26524#discussion_r185150957) [](ancestors = 185150957)

File: src/Compilers/CSharp/Test/Syntax/Parsing/PatternParsingTests.cs (hunk 1 above)
Comment (gafter) [reply src/Compilers/CSharp/Test/Syntax/Parsing/PatternParsingTests.cs]: Yes; adding a test.

---
In reply to: [185264756](https://github.com/dotnet/roslyn/pull/26524#discussion_r185264756) [](ancestors = 185264756)

**Comment by gafter** [comment]
@agocke Does this last delta address your concerns?

**Comment by jcouv** [comment]
@gafter Can this be merged? I plan to adjust some IDE logic following this change. Thanks

**Pull request merged**